src/descriptor/satisfied_constraints.rs

```diff
@@ -1632,9 +1642,9 @@ mod tests {
         //Check ThresM
         let stack = vec![
             StackElement::Dissatisfied,
-            StackElement::Push(&der_sigs[0]),
-            StackElement::Push(&der_sigs[1]),
             StackElement::Push(&der_sigs[2]),
+            StackElement::Push(&der_sigs[1]),
+            StackElement::Push(&der_sigs[0]),
```

Comment:
I remember checking out the bitcoin core interpreter.cpp but still got this messed up.